ts/sagas/startup.ts

```diff
@@ -160,11 +164,15 @@ function* initializeApplicationSaga(): IterableIterator<Effect> {
     if (!isSessionRefreshed) {
       // The user was previously logged in, so no onboarding is needed
       // The session was valid so the user didn't event had to do a full login,
-      // in this case we ask the user to provide the PIN as a "lighter" login
-      yield race({
-        login: call(loginWithPinSaga, storedPin),
-        reset: call(watchPinResetSaga)
-      });
+      // in this case we ask the user to identify using the PIN.
+      const identificationResult: SagaCallReturnType<
+        typeof startAndReturnIdentificationResult
+      > = yield call(startAndReturnIdentificationResult, storedPin);
+      if (identificationResult !== IdentificationResult.success) {
```

Comment:
should't check for `=== IdentificationResult.pinreset` ?

Reply:
I think any result different from success must restart the application (in the future we'll have also fingerprintreset for example).

Reply:
yes, in fact you should enumerate all results that trigger a restart... or else, by looking at this code, I can't understand under which circumstances the flow gets restarted